node/bft/src/primary.rs

```diff
@@ -522,14 +522,39 @@ impl<N: Network> Primary<N> {
                     }
                     // Check the transmission is still valid.
                     match (id, transmission.clone()) {
-                        (TransmissionID::Solution(solution_id), Transmission::Solution(solution)) => {
+                        (TransmissionID::Solution(solution_id, checksum), Transmission::Solution(solution)) => {
```

Comment:
I think these checksum checks are superfluous. We already confirm the checksum when we receive the transmission. And in contrast to some other added superfluous checks, there was no precedent to check if the solution_id was consistent. So I suggest removing these.

Reply:
The subsequent calls to `check_solution_basic` and `check_transaction_basic` do enforce that the `{transaction/solution}ID` does match the one in the object. So for consistency, I added the checksum enforcement here as well.

Reply:
I see, how about moving the checksum check into `snarkOS::check_{transaction, solution}_basic`? I see it would require changing the `LedgerService` trait slightly by passing in TransmissionId or the checksum argument. What we gain is if we ever want to remove those checks for improved speed/readability, its in one place.

Reply:
I thought about that and decided against for a few reasons:

1. Keep code change as minimal as possible.
2. The entry points for `check_*_basic` is usually via raw transactions or solutions from peer messaging rather than via transmissions, so adding a checksum check here may actually be more expensive overall.
3. Optimizations can be added in the future, the current priority is safety and simplicity. Especially since the checksum checks here are fairly cheap.

LMK if that makes sense or there are holes in my logic.